django_mongodb/schema.py

```diff
@@ -122,11 +148,43 @@ def alter_index_together(self, model, old_index_together, new_index_together):
             self._add_composed_index(model, field_names)
 
     def alter_unique_together(self, model, old_unique_together, new_unique_together):
-        pass
+        olds = {tuple(fields) for fields in old_unique_together}
+        news = {tuple(fields) for fields in new_unique_together}
+        # Deleted uniques
+        for field_names in olds.difference(news):
+            self._remove_composed_index(
+                model,
+                field_names,
+                {"unique": True, "primary_key": False},
+            )
+        # Created uniques
+        for field_names in news.difference(olds):
+            columns = [model._meta.get_field(field).column for field in field_names]
+            name = str(self._unique_constraint_name(model._meta.db_table, columns))
+            constraint = UniqueConstraint(fields=field_names, name=name)
+            self.add_constraint(model, constraint)
 
-    def add_index(self, model, index, field=None):
+    def add_index(self, model, index, field=None, unique=False):
         if index.contains_expressions:
             return
+        kwargs = {}
+        filter_expression = defaultdict(dict)
+        if index.condition:
+            filter_expression.update(index._get_condition_mql(model, self))
+        if unique:
+            kwargs["unique"] = True
+            if field:
+                filter_expression[field.column].update({"$type": field.db_type(self.connection)})
```

Comment:
The comment is below: "Use partialFilterExpression to allow multiple null values for a unique constraint." It's not really in the best spot and should be amended to reference `$type` since partialFilterExpression is also used for `condition`.